### PR TITLE
get source: improve instructions

### PR DESCRIPTION
**Pull request by n0toose**
- Commands can now be copied and pasted on a variety of systems using user environment variable. As this is not a catch-all solution, a sufficient warning was added.
- Replace example domains with "example.com"
- Improve some headers
- Add some additional background information for better onboarding
- Add additional warning for GitHub pull requests
- Fix English
- Make the tone more informal as to not intimidate new contributors

### CODE REVIEW

**Comment by n0toose** [comment]
I spent a lot of time revisiting this again, but my changes may contain some trivial errors or mistakes.

File: content/guides/building/get-source-git.md [hunk 1]
```diff
@@ -48,64 +48,75 @@ to the last 10 commits, for example.
...
+<strong>Configure your Git!</strong> Before making any commits, make <strong>sure</strong> that you have <a href="#configure_env">configured your Git environment</a> properly! Failing to configure Git properly before a commit will result in incorrect naming in your commit, making it impossible to give you well-deserved credit for your work.</div>
 
 <h4>Configure Git on your system:<a name="configure_env"></a></h4>
 
-Before making your first commit on a new system, be <strong>sure</strong> to configure Git. These global settings are stored in your git configuration directory (`~/.git/` or for Haiku: `~config/settings/git/`) and will be appended to <strong>each</strong> commit as your personal information.
+Using `--global` will your personal information in your Git configuration directory (`~/.git/` or for Haiku: `~config/settings/git/`), which will be included in <strong>every</strong> commit.
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:55]:
```suggestion
Using `--global` will your personal information in your Git configuration directory (`~/.git/` or for Haiku: `~/config/settings/git/`), which will be included in <strong>every</strong> commit.
```

File: content/guides/building/get-source-git.md [hunk 2]
```diff
@@ -48,64 +48,75 @@ to the last 10 commits, for example.
...
 ```
 
 <h4>Setup an account on Gerrit</h4>
 
-<p>Log in to <a href="https://review.haiku-os.org">Gerrit code review</a>. You currently need a <a href="https://github.com">Github</a> account for logging in.</p>
+<p>Log in to <a href="https://review.haiku-os.org">Gerrit code review</a>. For the time being, using a <a href="https://github.com">GitHub account</a> is necessary.</p>
 
 <div class="alert alert-warning">
-Make sure your ssh key is generated with ed25519. RSA keys will not work with Gerrit.
+Make sure your SSH key is generated using <code class="code">ed25519</code>. RSA keys will <b>not</b> work with Gerrit.
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:73]:
```suggestion
Make sure your SSH key is generated using <code class="code">ed25519</code>. RSA keys will <strong>not</strong> work with Gerrit.
```

(for consistency with the other warnings in the page)

File: content/guides/building/get-source-git.md [hunk 3]
```diff
@@ -48,64 +48,75 @@ to the last 10 commits, for example.
...
 </div>
-<p>Upload your SSH public key in <a href="https://review.haiku-os.org/settings/#SSHKeys">Gerrit SSH keys settings</a> page. If you don't have a key yet, you can generate one using ssh-keygen.</p>
+<p>Upload your SSH public key in the <a href="https://review.haiku-os.org/settings/#SSHKeys">Gerrit SSH Keys settings</a> page. If you don't have a key yet, you can generate one using ssh-keygen.</p>
 <p>If the e-mail address used in your commits does not match the one in your github account, you will need to add and verify it in <a href="https://review.haiku-os.org/settings/#EmailAddresses">Gerrit E-Mail address settings</a>.</p>
 
-<h4>Build Tools:</h4>
+<div class="alert alert-warning">
+You may need to replace the <code class="varname">$USER</code> parameter in the following commands.
+</div>
+<p>This can happen if the username on your Gerrit account does not match the username that you use to log in to your computer. This would be the case in Haiku itself, as the default username is <code class="code">user</code>. You can make your life easier by <a href="https://linux.die.net/man/5/ssh_config">modifying your SSH configuration</a>.</p>
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:81]: A link to a man page documenting the whole file is not very helpful. Especially in this page which is supposed to be an "all-in-one" getting started tutorial.

File: content/guides/building/get-source-git.md [hunk 4]
```diff
@@ -48,64 +48,75 @@ to the last 10 commits, for example.
...
+<p><code class="varname">$USER</code> is an environment variable that is specific to Unix-like operating systems (e.g. Haiku, Linux, FreeBSD, etc.). If your system does not fall under that category, you will probably have to replace it.</p>
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:82]: Well, if that does not work all the time (and especially not with the default Haiku setup), maybe we should not attempt to use it. Let's keep this page as short as possible.

If you want to make things explicit in the page, maybe do something like:

`export GERRIT_USERNAME=john_doe`

(keeping with the user that was used as an example earlier)

File: content/guides/building/get-source-git.md [hunk 5]
```diff
@@ -48,64 +48,75 @@ to the last 10 commits, for example.
...
+<h4>Sending in your first patch</h4>
+
+So, you want to contribute to Haiku. That's awesome, thank you for your interest!
 
-Want to submit your first patch? Thanks, that's great! You don't need to checkout the sources again. Instead you can update your existing copy of the source to use the commiter access. Just change the remote URL:
+<div class="alert alert-warning">
+Pull requests for the Haiku operating system made using GitHub cannot be accepted.
+</div>
+
+If you only have experience with GitHub and downloaded the source code of Haiku from it, don't worry, we've got you.
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:104]: I don't think this adds anything relevant to the page and the tone seems a bit too familiar.

There is already a section a bit further down documenting the Gerrit flow for people familiar with Github.

File: content/guides/building/get-source-git.md [hunk 6]
```diff
@@ -48,64 +48,75 @@ to the last 10 commits, for example.
...
 
-The `<login>@` is only needed if your currently logged in username doesn't match your `review.haiku-os.org` username.
-
 ```sh
-git clone "ssh://<login>@git.haiku-os.org/haiku"
-scp -p <login>@git.haiku-os.org:hooks/commit-msg "haiku/.git/hooks/"
+git clone "ssh://$USER@git.haiku-os.org/haiku" && scp -p $USER@git.haiku-os.org:hooks/commit-msg "haiku/.git/hooks/"
 ```
 
-<h4>Switching from read-only to write access</h4>
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:95]: this section was clearly labelled and this title made sense if following the previous paragraphs.

Now this info is removed, and there is no context for the commands below.

File: content/guides/building/get-source-git.md [hunk 7]
```diff
@@ -48,64 +48,75 @@ to the last 10 commits, for example.
...
 
-You also should install the <a href="https://review.haiku-os.org/Documentation/user-changeid.html">Gerrit hooks to generate Change-Ids</a>:
+You also should install a Git hook that will automatically <a href="https://review.haiku-os.org/Documentation/user-changeid.html">generate Change-Id's</a> for you:
 
 ```sh
-scp -p <login>@git.haiku-os.org:hooks/commit-msg "haiku/.git/hooks/"
+scp -p $USER@git.haiku-os.org:hooks/commit-msg "haiku/.git/hooks/"
 ```
 
+You should definitely do that, by the way. If a maintainer asks you to correct something later on, Gerrit will use that Change-Id to keep track of your newest changes.
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:118]: here as well I think the tone is a bit too familiar. Also, we should really just make this mandatory in Gerrit config, or update our Gerrit to a version that's not broken where the Change-Id is visible on the server side when Gerrit generates it.

File: content/guides/building/get-source-git.md [hunk 8]
```diff
@@ -116,16 +127,16 @@ Haiku's source code needs to reside on a case sensitive file system.
...
 <a name="proxy_access"></a>
-<h4>Getting the source code through an HTTP proxy</h4>
+<h4>Obtaining the source code using an HTTP proxy</h4>
 
 <div class="alert alert-warning">
 Haiku's main Git repository does not allow HTTP access, which is a problem if you are accessing the Internet through a proxy server that only permits HTTP (port 80) traffic.
 </div>
 
-Instead, use one of our mirror repositories at GitHub or Gitorious for anonymous HTTP access, they are both kept in sync with the main repository. First, set Git to connect through your proxy server:
+Use one of our mirror repositories <a href="https://github.com/haiku/haiku">at GitHub</a> for anonymous HTTP access instead. They are both kept in sync with the main repository. First, set Git to connect through your proxy server:
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:136]:
```suggestion
Use our mirror repositories <a href="https://github.com/haiku/haiku">at GitHub</a> for anonymous HTTP access instead. They are kept in sync with the main repository. First, set Git to connect through your proxy server:
```

File: content/guides/building/get-source-git.md [hunk 9]
```diff
@@ -137,6 +148,8 @@ git clone http://github.com/haiku/haiku.git
 
 Please note that these repositories do not contain any `hrev` tags, which are used by the Haiku build system to determine the Haiku revision. To work around this limitation, use the <a href="https://cgit.haiku-os.org/haiku/tree/build/jam/UserBuildConfig.ReadMe" target="_blank">HAIKU_REVISION build variable</a> when building Haiku.
 
+The limitation is caused by the fact that GitHub does not support repositories with tens of thousands of tags very well. A new tag is created once every few days and used to create our <a href="https://download.haiku-os.org">Nightly images</a>.
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:151]: This is incorrect. Tags are created whenever something is pushed to the master branch, not every few days (it's closer to multiple times a day). And "every few days" wouldn't make sense for nightly images.

Anyway, this kind of technical details should probably not be included in this page. Knowing that it won't work and how to fix/workaround it is enough here. Technical details should go somewhere else?

File: content/guides/building/get-source-git.md (hunk 4 above)
Comment (n0toose) [inline content/guides/building/get-source-git.md:82]: Should this be removed? The platforms that we support are explicitly stated here: https://github.com/haiku/haiku/blob/07f11ccca916aa828439b25d51713f1cf610b79a/ReadMe.Compiling.md#building-haiku, however, the document does not mention this as it is out of scope.

However, this is the first thing people should supposedly read before they compile Haiku. How do I work around that?

File: content/guides/building/get-source-git.md (hunk 6 above)
Comment (n0toose) [reply content/guides/building/get-source-git.md:95]: This was accidental.

File: content/guides/building/get-source-git.md (hunk 7 above)
Comment (n0toose) [reply content/guides/building/get-source-git.md:118]: > Also, we should really just make this mandatory in Gerrit config.

My changes here were really based after real mistakes I made myself, this is off-topic but you should absolutely go for that.

File: content/guides/building/get-source-git.md (hunk 8 above)
Comment (n0toose) [reply content/guides/building/get-source-git.md:136]: What other mirror repositories are there?

File: content/guides/building/get-source-git.md (hunk 4 above)
Comment (pulkomandy) [reply content/guides/building/get-source-git.md:82]: Really this should be all moved to https://www.haiku-os.org/docs/develop/ where we can have a book-like thing, with pages you have to read in order. The website does not provide that, I think, unless we start manually adding a kind of "next/previous page" link somewhere?

Maybe add yet another warning to this page linking to the list of supported platforms.

File: content/guides/building/get-source-git.md (hunk 9 above)
Comment (n0toose) [reply content/guides/building/get-source-git.md:151]: > This is incorrect. Tags are created whenever something is pushed to the master branch, not every few days (it's closer to multiple times a day). And "every few days" wouldn't make sense for nightly images.

I remembered a conversation we had because of this subject, I'd definitely go "huh, why" and there's no other designated place for that short detail.

File: content/guides/building/get-source-git.md (hunk 3 above)
Comment (n0toose) [reply content/guides/building/get-source-git.md:81]: Understandable. Do I just give the reader a sample ~/.ssh/config and tell them the directories to put them in? (e.g. `~/.ssh/config` on Linux, `/boot/system/settings/ssh/sshd_config` on Haiku.

File: content/guides/building/get-source-git.md (hunk 7 above)
Comment (n0toose) [reply content/guides/building/get-source-git.md:118]: > You should definitely do that, by the way.

is indeed way too direct and friendly, but should the "you" pronoun/addressing the reader be done, in your opinion?

File: content/guides/building/get-source-git.md (hunk 9 above)
Comment (pulkomandy) [reply content/guides/building/get-source-git.md:151]: The answer to "why" is mainly "stupid historical reasons when we switched from SVN".

We could use "git describe --tags" or similar and our life would be a lot simpler. But back when we switched to git, there was no such thing or it was too slow to be usable.

But there could be a page or two of text explaining how the hrevs (and btrevs) work, how they are generated, why they are needed, what problems they create, etc. Then it's not "just a short detail" anymore AND we can provide accurate info :)

Comment (n0toose) [reply content/guides/building/get-source-git.md:151]: *misremembered, my bad, I just started contributing again. :D

File: content/guides/building/get-source-git.md (hunk 4 above)
Comment (n0toose) [reply content/guides/building/get-source-git.md:82]: Let's just remove the warning for the time being and hope that the user won't use Windows?

Comment (n0toose) [reply content/guides/building/get-source-git.md:82]: P.S. That's a story for another day, but it's a bit hard to move the documents, as they all have to be moved at once and I am not sure what's the standard practice on introducing redirects. (I know where to put them, but has this been done before? It's why I never got around to it.

Comment (pulkomandy) [reply content/guides/building/get-source-git.md:82]: we had one confused user try this under WSL2, and it fails only much later. So it would be nice to keep the warning.

Comment (n0toose) [reply content/guides/building/get-source-git.md:82]: > we had one confused user try this under WSL2, and it fails only much later.

good god

File: content/guides/building/get-source-git.md (hunk 3 above)
Comment (pulkomandy) [reply content/guides/building/get-source-git.md:81]: If you set the login in the URL as we already recommend, I think the ssh-config is not needed. So it seems simpler to just not mention it at all here?

File: content/guides/building/get-source-git.md (hunk 4 above)
Comment (pulkomandy) [reply content/guides/building/get-source-git.md:82]: I think the simplest way to proceed here is not try to rely on $USER being already set to the right thing.

So let's set it manually:

    set GERRIT_USER=your gerrit username
    git clone $GERRIT_USER@...

And maybe add a screenshot showing where to find the username in Gerrit

File: content/guides/building/get-source-git.md (hunk 8 above)
Comment (pulkomandy) [reply content/guides/building/get-source-git.md:136]: I'm not aware of other mirrors, that's why I just removed the "both" here which referred to github and gitorious before.

Comment (n0toose) [reply content/guides/building/get-source-git.md:136]: > Use one of our mirror repositories

... is still there, I'll commit the change and fix this later.

File: content/guides/building/get-source-git.md (hunk 6 above)
Comment (n0toose) [reply content/guides/building/get-source-git.md:95]: Wait, no, "write access" insinuates "commit access", which is why I corrected that header and replaced it with `<h4>Sending in your first patch</h4>`.

**Comment by n0toose** [comment]
Oops, I seem to have been a bit careless with my rebasing as this is a PR. Sorry about that, fixing in a sec.

**Comment by n0toose** [comment]
Fixing few more minor mistakes...

**Comment by n0toose** [comment]
I rebased the changes made in this repository, added `target=_blank` consistently and "butchered" the entire proxy guide and included the anonymous HTTP(S) stuff in the main article (and removed any outdated mentions to just a port 80) so as to make the structure of the document more linear and easy to read. I removed the proxy instructions. I changed the wording of a few additional sentences.

I believe that there's a few factual errors in the piece itself, which I am trying to amend.

**Comment by n0toose** [comment]
Okay, there's still a lot of stuff to fix. Need some more time.

**Comment by n0toose** [comment]
image

This is factually incorrect, isn't it?

**Comment by pulkomandy** [comment]
> image
> 
> This is factually incorrect, isn't it? (because of `review.haiku-os.org`)

It used to be a problem, the HTTP cloning would timeout. I think this has since been solved.

**Comment by n0toose** [comment]
> It used to be a problem, the HTTP cloning would timeout. I think this has since been solved.

Fixed. I just pushed some major changes that changes the entire structure of the document in a way that is linear (and allows for it to be split in two parts later, which was the reason why I claimed that it was not appropriate to include it in `docs/` just yet). I think it looks as good as I want it to be, but I am not too sure about e.g. the way I've placed the overview.

File: content/guides/building/get-source-git.md [hunk 10]
```diff
@@ -9,25 +9,29 @@ tags = []
...
 <h3>Git Access - Anonymous testers</h3>
 
 <div class="alert alert-warning">
 <strong>Anonymous access is read-only.</strong> If you want to submit changes to
-Haiku, you need to follow the instructions for patch submitters, in the next
+Haiku, you need to follow the instructions for patch submitters in the next
 section.
 </div>
 
+<div class="alert alert-warning">
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:30]:
```suggestion
<div class="alert alert-info">
```

File: content/guides/building/get-source-git.md [hunk 11]
```diff
@@ -40,141 +44,150 @@ git clone https://review.haiku-os.org/buildtools
...
+<img src='/files/gitProcess_0.png' alt='A diagram describing the way our Git workflow works. This picture is there for aesthetic purposes and can be skipped. After a repository is cloned locally, if you have an idea, you can convert that idea into a commit using food (a pizza is depicted in the diagram). That commit will receive some change proposals, which will result in some further changes. The circle continues until the change is ready to be included in Haiku itself.'>
 
-```sh
-git clone "ssh://<login>@git.haiku-os.org/buildtools" && scp -p <login>@git.haiku-os.org:hooks/commit-msg "buildtools/.git/hooks/"
-```
+<h4>Configuring Git locally<a name="configure_env"></a></h4>
 
-<h4>Haiku:</h4>
+<div class="alert alert-danger">
+<strong>This section is important!</strong> Failing to configure Git properly before a commit will result in an incorrect name being used in your commit, making it impossible to give you well-deserved credit for your work.</div>
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:93]: all the sections are... I think having these warnings everywhere just makes the information less concentrated and harder to find.

File: content/guides/building/get-source-git.md [hunk 12]
```diff
@@ -40,141 +44,150 @@ git clone https://review.haiku-os.org/buildtools
...
+git config --global user.name "John Doe"
+git config --global user.email "john.doe@example.com"
 ```
 
-<h4>Switching from read-only to write access</h4>
-
-Want to submit your first patch? Thanks, that's great! You don't need to checkout the sources again. Instead you can update your existing copy of the source to use the commiter access. Just change the remote URL:
+Although most contributors tend to disclose their real names, we do not have
+a real name policy and plenty of contributors contribute to Haiku using a pseudonym.
+That means that it is possible for anyone to contribute.
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:104]: While this is an important topic to you, I don't think this is the right place to mention it in this way.

```suggestion
You are not required to use your "real" name, a pseudonym is fine and several Haiku contributors chose to do that. However, make sure the e-mail address is correct and working, it is what we will use if we need to contact you later.
```

File: content/guides/building/get-source-git.md [hunk 13]
```diff
@@ -40,141 +44,150 @@ git clone https://review.haiku-os.org/buildtools
...
 
-<h3>Some Notes</h3>
+<h4>Setup an account on Gerrit</h4>
 
-<h4>Case Sensitive Filesystem</h4>
+<p>Log in to <a href="https://review.haiku-os.org" target="_blank">Gerrit code review</a>. For the time being, using a <a href="https://github.com">GitHub account</a> is necessary.</p>
 
 <div class="alert alert-warning">
-Haiku's source code needs to reside on a case sensitive file system.
+Make sure your SSH key is generated using <code class="code">ed25519</code>. RSA keys will <strong>not</strong> work with Gerrit.
```
Comment (pulkomandy) [inline content/guides/building/get-source-git.md:118]: I would provide a working ssh-keygen example with the right options here. Saying "you can use ssh-keygen but make sure you get an ed25519 key" makes the tutorial not self-contained, and forces people to research ssh-keygen options elsewhere.

File: content/guides/building/get-source-git.md (hunk 12 above)
Comment (n0toose) [reply content/guides/building/get-source-git.md:104]: I honestly just tried to rewrite this, instead of impose my own biases and opinions on the subject. I am committing this for now, but I'll have to change the phrasing later.
